tere: init at 1.1.0

ProducerMatt opened the pull request:
###### Description of changes

A failing test was keeping it from building. I [confirmed with the
dev](https://github.com/mgunyho/tere/issues/44) it's known and ignored
so I commented it out with a patch.

This app isn't that useful until some [shell hooks have been
added.](https://github.com/mgunyho/tere#setup) So I hope to add a config
option to home-manager such as:

```nix
programs.tere = {
  enabled = true;
  useBashIntegration = true;
  useFishIntegration = true;
};
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

legendofmiracles commented on maintainers/maintainer-list.nix:
```diff
@@ -14748,4 +14748,10 @@
     github = "nikstur";
     githubId = 61635709;
   };
+  ProducerMatt = {
+    name = "Matthew Pherigo";
+    email = "ProducerMatt42@gmail.com";
+    github = "ProducerMatt";
+    githubId = 58014742;
+  };
```
File should be sorted somewhat alphabetically. Same goes for all-packages.nix

legendofmiracles commented on pkgs/tools/misc/tere/default.nix:
```diff
@@ -0,0 +1,34 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+let
+  commonMeta = {
+    name = "tere";
+    version = "1.1.0";
+  };
+  tereSrc = fetchFromGitHub {
+    owner = "mgunyho";
+    repo = commonMeta.name;
+    rev = "v${commonMeta.version}";
+    sha256 = "BD7onBkFyo/JAw/neqL9N9nBYSxoMrmaG6egeznV9Xs=";
+  };
+in
```
These are basically useless since you only use them once in the `in` block anyways ?

It's definitely cleaner to just remove these and inline

legendofmiracles commented on pkgs/tools/misc/tere/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+    # This test confirmed not working.
+    # https://github.com/mgunyho/tere/issues/44
+    cargoPatches = [ ./brokentest.patch ];
```
Instead of commenting everything out with // you can use /* at the beginning of the test, and at the end.
Or you can set `doCheck = false`, which will disable all tests.

legendofmiracles commented on pkgs/tools/misc/tere/default.nix:
```diff
@@ -0,0 +1,34 @@
+      homepage = "https://github.com/mgunyho/tere";
+      license = licenses.eupl12;
+      maintainers = [ maintainers.ProducerMatt ];
+      platforms = platforms.all;
```
```suggestion
```
set by buildRustPackage

ProducerMatt replied:
I like your suggestions. Should I rebase them into the original commit or add them as a separate commit?

zendo commented:
LGTM. Tested building and running fine.

winterqt reviewed (changes requested):
In addition to the suggestions you've already received, please split these changes into two commits:
1. adding yourself to the maintainer list (follow the naming scheme for other commits that update the list, i.e. `maintainers: add ProducerMatt`)
2. adding the package

ProducerMatt committed:
tere: init at 1.1.0

A failing test was keeping it from building. I [confirmed with the
dev](https://github.com/mgunyho/tere/issues/44) it's known and ignored
so I commented it out with a patch.

This app isn't that useful until some [shell hooks have been
added.](https://github.com/mgunyho/tere#setup) So I hope to add a config
option to home-manager such as:

```nix
programs.tere = {
  enabled = true;
  useBashIntegration = true;
  useFishIntegration = true;
};
```

ProducerMatt committed:
maintainers: add ProducerMatt

ProducerMatt commented:
I'm sorry that was the wrong branch. Hang on

ProducerMatt commented:
That just brought in a bunch of unrelated commiters didn't it? I'm really sorry. This happened earlier too, I'm obviously doing something wrong with my rebases. I'll try to make a new branch and cherry-pick since that worked earlier.

winterqt commented:
How are you rebasing (what command are you running)? Are you pointing it at a remote that points upstream?

ProducerMatt commented:
Ok let me know if this is all correct. Looks like the participating counter is back to normal.

> How are you rebasing (what command are you running)? Are you pointing it at a remote that points upstream?

When I started making suggested changes, I made a branch `tere3`. I made my changes, I did git rebase -i to squash them together, then I went to `tere2` and did `git reset --hard tere3` to make `tere2` identical. But I'm not sure I did anything like that last time it happened. Maybe it's because I didn't `git pull` before pushing?

winterqt commented:
Not sure honestly, kind of hard to reproduce with just that info. Looks good now, though.

ProducerMatt commented:
Yeah I'm not sure either. I'll have to put reading a book on Git at the top of my priority list. Sorry about the mess and thanks for the help. I'm ready when you guys are.

winterqt commented on pkgs/tools/misc/tere/default.nix:
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec
+  {
```
Please make the formatting look like this (see [here](https://github.com/NixOS/nixpkgs/blob/f768f101891cf5c40601ffeb66342f6478ed1d06/pkgs/tools/text/ripgrep/default.nix) to see how your formatting should look like (2 spaces instead of 4 for the keys)):

```suggestion
rustPlatform.buildRustPackage rec {
```

ProducerMatt replied:
Pushed, is this right?

winterqt replied:
Yup!

winterqt reviewed (changes requested):
Please swap the order of the two commits -- the maintainer addition commit should come first.

winterqt commented on pkgs/tools/misc/tere/default.nix:
```diff
@@ -0,0 +1,26 @@
+    description = "A faster alternative to cd + ls";
+    homepage = "https://github.com/mgunyho/tere";
+    license = licenses.eupl12;
+    maintainers = [ maintainers.ProducerMatt ];
```
```suggestion
    maintainers = with maintainers; [ ProducerMatt ];
```

winterqt reviewed (changes requested):
Last change :)

mgunyho commented:
Hi @ProducerMatt, the failing test has now been fixed with the release of `tere` v1.3.0. If you want, you can remove the patch now.

ProducerMatt commented:
Done https://github.com/NixOS/nixpkgs/pull/196177
